perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,714 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "autoware/tensorrt_bevdet/bevdet.hpp"
+
+#include "autoware/tensorrt_bevdet/alignbev_plugin.hpp"
+#include "autoware/tensorrt_bevdet/bevpool_plugin.hpp"
+#include "autoware/tensorrt_bevdet/gatherbev_plugin.hpp"
+#include "autoware/tensorrt_bevdet/preprocess_plugin.hpp"
+
+#include <thrust/functional.h>
+#include <thrust/sort.h>
+
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+
+using std::chrono::duration;
+using std::chrono::high_resolution_clock;
+
+BEVDet::BEVDet(
+  const std::string & config_file, int n_img, std::vector<Eigen::Matrix3f> _cams_intrin,
+  std::vector<Eigen::Quaternion<float>> _cams2ego_rot,
+  std::vector<Eigen::Translation3f> _cams2ego_trans, const std::string & onnx_file,
+  const std::string & engine_file)
+: cams_intrin(_cams_intrin), cams2ego_rot(_cams2ego_rot), cams2ego_trans(_cams2ego_trans)
+{
+  InitParams(config_file);
+  if (n_img != N_img) {
+    std::cerr << "BEVDet needs " << N_img << " images, but " << n_img << " images were given!"
+              << std::endl;
+  }
+  auto start = high_resolution_clock::now();
+
+  std::shared_ptr<int> ranks_bev_ptr = nullptr;
+  std::shared_ptr<int> ranks_depth_ptr = nullptr;
+  std::shared_ptr<int> ranks_feat_ptr = nullptr;
+  std::shared_ptr<int> interval_starts_ptr = nullptr;
+  std::shared_ptr<int> interval_lengths_ptr = nullptr;
+
+  InitViewTransformer(
+    ranks_bev_ptr, ranks_depth_ptr, ranks_feat_ptr, interval_starts_ptr, interval_lengths_ptr);
+  auto end = high_resolution_clock::now();
+  duration<float> t = end - start;
+  std::cout << std::fixed << std::setprecision(4)
+            << "InitVewTransformer cost time: " << t.count() * 1000 << " ms" << std::endl;
+
+  if (access(engine_file.c_str(), F_OK) == 0) {
+    std::cout << "Inference engine prepared." << std::endl;
```

Comment:
Use `RCLCPP_INFO_THROTTLE` instead of `std::cout`

Reply:
When to use `RCLCPP_INFO` and when to use `std:: cout`, it's really confusing. If possible, could you help me change it directly.

Reply:
`RCLCPP_INFO` is suitable for logging tool in autoware instead of `std::cout`.

Reply:
Sorry, perhaps my statement has caused you a misunderstanding. What I meant was that there are many `std:: cout` and `std:: cerr` in the code, should we use `rclcpp_logger` instead?
 Is this necessary? The workload of making changes is also significant.

Reply:
The initial review comments should not have made me change `printf` to `std::out` or `std::cerr`, but `rclcpp_logger`. now, I have to change it again and again.

Reply:
Which review comment do you talk about ? :thought_balloon: 

Whatever the reason, as a reviewer, I have to say, "If you are conscious of the quality of autoware, you should use `RCLCPP` instead of `std::cout`." Please be patient.

Reply:
Thanks for your patient review.  To make Autoware better, I should be patient.
Some of the review comments are not easy for me to modify, and I don't have enough time to handle this PR. If you could help me revise them according to your suggestions during the review, I would be very grateful. 🙏🙏🙏🙏🙏

Reply:
> Thanks for your patient review. To make Autoware better, I should be patient.

Thank you for your cooperation! :+1: 

> I don't have enough time to handle this PR

I'm sorry to hear that you don't have enough time...
OK, let's update this code quickly :muscle: